Either/Bifunctor

```diff
@@ -0,0 +1,15 @@
+    let Bifunctor = ./../Bifunctor/Type
+
+in  let Either = ./Type
+
+in    { bimap =
+            λ(a : Type)
+          → λ(b : Type)
+          → λ(c : Type)
+          → λ(d : Type)
+          → λ(f : a → b)
+          → λ(g : c → d)
+          → λ(e : Either a c)
+          → ./mapBoth a c b d f g e
```

Comment:
Oh, I am clearly bad at the alphabet 😆

Ok, I think I’d be inclined to switch the type parameters of `bimap`, with this rationale – `Bifunctor` is a functor from a product category (with objects `(Type, Type)`), to the category of Dhall types. Since the specific objects we’re talking about in the product category here are `(a, c)` and `(b, d)`, it seems like those type parameters should be adjacent to each other.

If this were Haskell or something where we had implicits, I would be inclined to just inline `mapBoth` and delete that file. Since it can be a bit more verbose to get an instance in Dhall, maybe that’s not the best course of action.

Reply:
So I took the naming of the type variables form [Haskell's bimap](https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Bifunctor.html#v:bimap).

Should I change the order of the type arguments in `bimap`: `\a c b d ...`?
Or rename the type variables so I that it goes from `p a b` to `p c d`?

Your comment about inlining `mapboth` made me realise that I haven't added the Bifunctor instance for Either to `Either/package.dhall` yet.

Reply:
Yeah, I’d go with `p a b` and `p c d`.

Ah right … you know, since you’re adding Bifunctor to the Either/package, then I’m inclined to go back and say yeah – delete `mapBoth`.

@FintanH?

Reply:
I'll wait to hear @FintanH's point of view and can implement the changes tomorrow. Good night! :smile:

Reply:
Good morning! So I guess the two alternatives are:

1. 
```
  λ(p : Type → Type → Type)
→ { bimap :
        ∀(a : Type)
      → ∀(b : Type)
      → ∀(c : Type)
      → ∀(d : Type)
      → (a → b)
      → (c → d)
      → p a b
      → p c d
  }
```

2.
```
  λ(p : Type → Type → Type)
→ { bimap :
        ∀(a : Type)
      → ∀(c : Type)
      → ∀(b : Type)
      → ∀(d : Type)
      → (a → b)
      → (c → d)
      → p a c
      → p b d
  }
```

Option C) Leaving it as it is.

I think option 1. has the best layout since you can fully saturate the first type which is useful for currying this function, but it also is in alphabetical order so easier to grep. Option 2. suffers from the weird, non-alphabetical ordering. Option C. seems the weakest because you have to define the first param to the second `p` before saturating the first `p`.

My vote is option 1.

Reply:
For option 1, the two functions need to be `a -> c` and `b -> d`.
I think option 1 (with this extra fix) is the better option. I'll implement it, and push it here.
When you're happy with all the changes, I can squash my commits together! :smile: